cell.go

```diff
@@ -505,41 +505,38 @@ func (f *File) GetCellRichText(sheet, cell string) (runs []RichTextRun, err erro
 	if err != nil {
 		return
 	}
-
-	siIndex, err := strconv.Atoi(cellData.V)
+	siIdx, err := strconv.Atoi(cellData.V)
 	if nil != err {
 		return
 	}
-
 	sst := f.sharedStringsReader()
-	a := sst.SI[siIndex]
-	for _, v := range a.R {
+	if len(sst.SI) <= siIdx || siIdx < 0 {
+		return
+	}
+	si := sst.SI[siIdx]
+	for _, v := range si.R {
 		run := RichTextRun{
 			Text: v.T.Val,
 		}
 		if nil != v.RPr {
 			font := Font{}
-			font.Bold = v.RPr.B == " "
-			font.Italic = v.RPr.I == " "
+			font.Bold = v.RPr.B != nil
+			font.Italic = v.RPr.I != nil
 			if nil != v.RPr.U {
 				font.Underline = *v.RPr.U.Val
```

Comment:
underline has two formats `<u/>` and `<u val="single"></u>`, so it should check `v.RPr.U.Val`
```
if nil != v.RPr.U {
	if nil != v.RPr.U.Val {
		font.Underline = *v.RPr.U.Val
	} else {
		// use single for default
		font.Underline = "single"
	}
}
```